### PR TITLE
feat: compress pair values of certain rule types

### PR DESCRIPTION
**Pull request by nsrCodes**
compresses the pair value for 
1. Response rule
2. Request rule
3. Script rule that are of code type

in transit (to and from firebase) and storage. 
Syncing still works on the complete (uncompressed) records

### CODE REVIEW

**Comment by sagarsoni7** [comment]
We’re first compressing the rule data and then encoding it to base64.
Wouldn’t encoding to base64 increase the size again?
If we are doing base64 encoding to convert it into text to set data in Realtime db, how about directly pushing the compressed outcome to Firestore DB instead? It accepts ByteString

For now, we can move forward with RealtimeDB but we still have the future scope to compress further by moving rules to FirestoreDB right? Of course Firestore has its own challenges we know

**Review by sagarsoni7 — APPROVED** [review]

File: app/src/utils/Compression.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+          pair.scripts.forEach((script, sidx) => {
+            if (script.type === "code") {
+              compressedRecord.pairs[idx].scripts[sidx].value = compressString(script?.value ?? "");
+              compressedRecord.pairs[idx].scripts[sidx].isCompressed = true;
+            }
+          });
+        });
+        break;
+      default:
+        console.log("This rule type should not need compression");
```
Comment (wrongsahil) [inline app/src/utils/Compression.js:106]: Remove log

File: app/src/utils/Compression.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        console.log("This rule type should not need decompression");
+    }
+  }
+  return decompressedRecord;
+}
+
+export function decompressRecords(recordsMap) {
+  const decompressedRecordsMap = {};
+  for (const recordId in recordsMap) {
+    if (shouldDecompressRecord(recordId)) {
```
Comment (wrongsahil) [inline app/src/utils/Compression.js:66]: This looks redundant as this check is already present in decompressRecords.

File: app/src/utils/Compression.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+function shouldDecompressRecord(recordId) {
+  // trying to keep it as efficient as possible by working with ID instead of whole object
+  const recordType = recordId.split("_")[0];
+  return recordType && ["Response", "Request", "Script"].includes(recordType);
+}
+
+function decompressRecord(record) {
+  const decompressedRecord = { ...record };
+  if (isRule(record)) {
```
Comment (wrongsahil) [inline app/src/utils/Compression.js:25]: Missing Check for `isCompressed` key

**Review by wrongsahil — CHANGES_REQUESTED** [review]

File: app/src/utils/Compression.js (hunk 3 above)
Comment (nsrCodes) [reply app/src/utils/Compression.js:19]: this is handled below on the pair level

**Review by wrongsahil — APPROVED** [review]

**Pull request merged**